src/classes/EP_ManageEPTemplate_CTRL.cls

```diff
@@ -0,0 +1,629 @@
+                //don't namespace prefix standard fields like Name
+                fieldName.endsWith('__c')?UTIL_Namespace.StrTokenNSPrefix(fieldName):fieldName
+            ), 
+            styleClass='slds-form-element__label'
+        );
+        fieldLabel.for = fieldName + wrapper.level;
+        result.childComponents.add(fieldLabel);
+        return result;
+    }
+
+    /** @description Generates the markup for an SLDS styled apex:inputField.
+    * @param wrapper The TaskWrapper for the current Engagement Plan Task.
+    * @param accessString The string for accessing the current place in the heirarchy using a nested expression.
+    * @param fieldName The name of the field.
+    * @param outterCss CSS markup for the field input container.
+    * @param inputCss CSS markup for the input itself.
+    * @return Component.Apex.OutputPanel OutputPanel containing the markup for an inputField.
+    */
+    private Component.Apex.OutputPanel genericInputPanel(TaskWrapper wrapper, String accessString, String fieldName, String outterCss, String inputCss) {
+        Component.Apex.OutputPanel outterPanel = fieldLabelPanel(wrapper, fieldName, outterCss);
+
+        Component.Apex.InputField fieldInput = new Component.Apex.InputField(
+            id = fieldName + wrapper.level,
+            value = wrapper.detail.get(fieldName),
+            styleClass = 'slds-form-element__control ' + inputCss,
+            //Set the Name field as required for auto validation
+            required = fieldName == 'Name'
+        );
+        fieldInput.expressions.value = '{!' + accessString + '.detail.' + fieldName + '}';
+        outterPanel.childComponents.add(fieldInput);
+
+        return outterPanel;
+    }
+
+    /** @description Generates the markup for an SLDS styled picklist.
+    * @param wrapper The TaskWrapper for the current Engagement Plan Task.
+    * @param accessString The string for accessing the current place in the heirarchy using a nested expression.
+    * @param fieldName The name of the field.
+    * @param outterCss CSS markup for the field input container.
+    * @param inputCss CSS markup for the input itself.
+    * @param selectOptions The controller parameter containing the selectOptions for this picklist.
+    * @return Component.Apex.OutputPanel OutputPanel containing the markup for a picklist field.
+    */
+    private Component.Apex.OutputPanel selectListInputPanel(TaskWrapper wrapper, String accessString, String fieldName, String outterCss, String inputCss, String selectOptions) {
+        Component.Apex.OutputPanel outterPanel = fieldLabelPanel(wrapper, fieldName, outterCss);
+
+        Component.Apex.SelectList selectList = new Component.Apex.SelectList(
+            id = fieldName + wrapper.level,
+            value = wrapper.detail.get(fieldName),
+            styleClass = 'slds-form-element__control ' + inputCss,
+            style = 'height: calc(1.875rem + 2px);',
```

Comment:
why not use class names for some css you put in the header of the page, rather than embedding this specific height calculation in the code.